Now builds on Ubuntu

ssoper opened the pull request:
Wasn't quite sure what to do about the `NSException`s so I converted them to `assertionFailure`. We could always branch on `os(Linux)`.

ssoper committed:
Now builds on Ubuntu

ssoper commented:
For the lib to be compatible with Swift package manager you'd need to do your release tags without the `v` prefix.
https://github.com/ssoper/Safe/releases/tag/0.1.0

tidwall commented:
I made some minor updates to get the unit tests working in xcode and merged the changes.
Thank a ton for putting this together!

tidwall commented:
Also could you double check in your Ubuntu environment and I'll tag a new release without the `v`.

ssoper commented:
Still builds in Ubuntu. The reason I changed `anyGenerator` was due to this warning

``` bash
/Users/ssoper/workspace/Safe/Source/chan.swift:117:16: warning: 'anyGenerator' is deprecated: renamed to 'AnyGenerator'
        return anyGenerator {
               ^
/Users/ssoper/workspace/Safe/Source/chan.swift:117:16: note: use 'AnyGenerator' instead
        return anyGenerator {
               ^~~~~~~~~~~~
               AnyGenerator
```
